metricshub-web/react/src/components/config/Tree/ConfigTree.jsx

```diff
@@ -15,27 +23,101 @@ const ROOT_ID = "__config_root__";
  * @returns {JSX.Element} The configuration tree component.
  */
 export default function ConfigTree({ files, selectedName, onSelect, onRename, onDelete }) {
+	const dispatch = useAppDispatch();
+	const list = useAppSelector((s) => s.config.list);
 	const selectedIds = React.useMemo(() => (selectedName ? [selectedName] : []), [selectedName]);
 
+	// config root kebab
+	const [rootMenuAnchor, setRootMenuAnchor] = React.useState(null);
+	const openRootMenu = (e) => {
+		e.stopPropagation();
+		setRootMenuAnchor(e.currentTarget);
+	};
+	const closeRootMenu = () => setRootMenuAnchor(null);
+	const handleBackupAll = async () => {
+		document.activeElement?.blur?.();
+		closeRootMenu();
+		try {
+			await dispatch(createConfigBackup({ kind: "all" })).unwrap();
+		} catch (e) {
+			console.error("Backup all failed:", e);
+		}
+	};
+
 	const folderLabel = (
```

Comment:
We should have only one component to prevent duplication of the left-side element here. Moreover, we should use React.memo too to avoid re render.
```jsx
const FolderLabel = React.memo(({ name, onMenuClick }) => (
  <Box sx={{ display: "flex", alignItems: "center", justifyContent: "space-between", width: "100%" }}>
    <Box sx={{ display: "flex", alignItems: "center" }}>
      <FileTypeIcon type="folder" />
      <span>{name}</span>
    </Box>
    {onMenuClick && (
      <IconButton size="small" onClick={onMenuClick}>
        <MoreVertIcon fontSize="small" />
      </IconButton>
    )}
  </Box>
));
```